src/resources/components/stats/stats.tsx

```diff
@@ -1,20 +1,55 @@
+import { useStorage } from "@plasmohq/storage/dist/hook";
+import { useEffect, useMemo, useState } from "react";
+import { getMetrics } from "~services/user/user-consumer-service";
+import type { AccessTokenResponse, MetricsResponse } from "~types/types";
+
 const Stats = () => {
-  const statsData = {
-    hoursWatched: 254,
-    messages: 420,
-    topChannels: [
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-    ],
-    topCategories: [
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-    ],
-  };
+  const [authorization] = useStorage<AccessTokenResponse>("accessToken");
+
+  const [stats, setStats] = useState<MetricsResponse>();
+
+  useEffect(() => {
+    getMetrics(authorization?.access_token)
+      .then((res) => {
+        console.log(res);
+        setStats(res);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, [authorization]);
+
+  const statsData = useMemo(() => {
+    if (!stats) {
+      return {
+        hoursWatched: 420,
+        messages: 420,
+        topChannels: [
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+        topCategories: [
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+      };
+    }
```

Comment:
Would it be better to set times and messages to 0 and leave channels and categories empty?

Reply:
Done.